src/demo-app/overlay/overlay-demo.ts

```diff
@@ -0,0 +1,45 @@
+import {Component, provide, ElementRef, ViewChildren, QueryList} from 'angular2/core';
+import {Overlay, OVERLAY_CONTAINER_TOKEN} from '../../core/overlay/overlay';
+import {ComponentPortal, Portal} from '../../core/portal/portal';
+import {BrowserDomAdapter} from '../../core/platform/browser/browser_adapter';
+import {TemplatePortalDirective} from '../../core/portal/portal-directives';
+
+
+@Component({
+  selector: 'overlay-demo',
+  templateUrl: 'demo-app/overlay/overlay-demo.html',
+  styleUrls: ['demo-app/overlay/overlay-demo.css'],
+  directives: [TemplatePortalDirective],
+  providers: [
+    Overlay,
+    provide(OVERLAY_CONTAINER_TOKEN, {useValue: document.body})
+  ]
+})
+export class OverlayDemo {
+  @ViewChildren(TemplatePortalDirective) templatePortals: QueryList<Portal<any>>;
+
+  constructor(public overlay: Overlay, public elementRef: ElementRef) {
+    BrowserDomAdapter.makeCurrent();
```

Comment:
Just a question: Why are you initializing the BrowserDomAdapater yourself?

Here is my post on Stackoverflow - [Click](http://stackoverflow.com/questions/34686856/angular-2-0-beta-access-dom-adapter-browserdomadapter)
And here's the Angular 2 Issue, which has been created from the stack: [Click](https://github.com/angular/angular/issues/6904)

As you can read in my stackoverflow post, the browser should initialize itself.